torch_frame/data/multi_nested_tensor.py

```diff
@@ -210,6 +220,95 @@ def _row_index_select(self, index: Tensor) -> Tensor:
         return MultiNestedTensor(num_rows=len(index), num_cols=self.num_cols,
                                  values=values, offset=offset)
 
+    def col_index_select(self, index: Tensor) -> 'MultiNestedTensor':
+        # TODO Implement
+        raise NotImplementedError
+
+    def single_index_select(self, index: int, dim: int) -> 'MultiNestedTensor':
+        r"""Get :obj:`index`-th row (:obj:`dim=0`) or column (:obj:`dim=1`)"""
+        if dim == 0 or dim == -3:
+            index = self._to_positive_index(index, dim=0)
+            start_idx = index * self.num_cols
+            end_idx = (index + 1) * self.num_cols + 1
+            offset = self.offset[start_idx:end_idx]
+            values = self.values[offset[0]:offset[-1]]
+            offset = offset - offset[0]
+            return MultiNestedTensor(num_rows=1, num_cols=self.num_cols,
+                                     values=values, offset=offset)
+        elif dim == 1 or dim == -2:
+            index = self._to_positive_index(index, dim=1)
+            start_idx = torch.arange(index, self.num_rows * self.num_cols,
+                                     self.num_cols, device=self.device)
+            diff = self.offset[start_idx + 1] - self.offset[start_idx]
+            batch, arange = batched_arange(diff)
+            # Compute values
+            values = self.values[self.offset[start_idx][batch] + arange]
+            # Compute offset
+            offset = diff.new_zeros(diff.numel() + 1)
+            torch.cumsum(diff, dim=0, out=offset[1:])
+            return MultiNestedTensor(num_rows=self.num_rows, num_cols=1,
+                                     values=values, offset=offset)
+        else:
+            raise RuntimeError(f"Unsupported dim={dim} for index_select.")
+
+    def _to_positive_index(
+        self,
+        index: Union[int, Tensor],
+        dim: int,
```

Comment:
There is a lot of duplicated logic here depending on `dim`. Might be easier to just add a `max_entries` argument here instead of `dim`?

Reply:
Good point. Done.